explainer.md

```diff
@@ -156,11 +154,21 @@ partial interface PresentationConnection {
 };
 
 dictionary PresentationSource {
-  readonly USVString url;
+  required PresentationSourceType type;
+
+  // Used if `type` is "url".
+  USVString? url;
```

Comment:
Should this also be readonly, as you're replacing a readonly `url` attribute on the PresentationConnection?

Reply:
Per the [spec](https://webidl.spec.whatwg.org/#idl-dictionaries):

> returning a dictionary from an operation will result in a new ECMAScript object being created from the current values of the dictionary. ... Modifications to the dictionary will not be reflected in the corresponding ECMAScript object, and vice-versa.

So it seems like marking dictionary attributes readonly wouldn't do much.